### PR TITLE
fix: the issue of custom-screensaver screen display

### PR DESCRIPTION
**Pull request by obslee**
When there is no picture in custom-screensaver's Settings directory, the screensaver should display black

Log: fix the issue of custom-screensaver screen display
Bug: https://pms.uniontech.com/bug-view-167341.html

### CODE REVIEW

File: customscreensaver/deepin-custom-screensaver/deepin-custom-screensaver.pro [hunk 1]
```diff
@@ -24,14 +24,15 @@ DEFINES += QT_DEPRECATED_WARNINGS
 
 CONFIG += c++11 link_pkgconfig
 
-PKGCONFIG += x11
+PKGCONFIG += x11 dtkwidget
```
Comment (Kakueeen) [inline customscreensaver/deepin-custom-screensaver/deepin-custom-screensaver.pro:27]: 新增了依赖，但是没有在control文件中添加依赖包

File: customscreensaver/deepin-custom-screensaver/src/slideshowscreensaver.cpp [hunk 2]
```diff
@@ -97,7 +92,67 @@ void SlideshowScreenSaver::paintEvent(QPaintEvent *event)
...
+    if (m_imagefiles.isEmpty())
+        return -1;
+
+    int index = 0;
+
+    if (m_shuffle) {
+        index = int(QRandomGenerator::global()->generate() % uint(m_imagefiles.count()));
+    } else {
+        if (index < m_imagefiles.count() - 1)
+            index++;
```
Comment (Kakueeen) [inline customscreensaver/deepin-custom-screensaver/src/slideshowscreensaver.cpp:127]: 该逻辑一直返回1

Comment (obslee) [reply customscreensaver/deepin-custom-screensaver/src/slideshowscreensaver.cpp]: done

File: customscreensaver/deepin-custom-screensaver/src/slideshowscreensaver.cpp [hunk 3]
```diff
@@ -61,19 +50,24 @@ SlideshowScreenSaver::~SlideshowScreenSaver()
 
 void SlideshowScreenSaver::onUpdateImage()
 {
-    if (m_imagefiles.isEmpty())
-        return;
+    if (!getCurrentImageIndex()) {
+
+        auto indexPath = m_imagefiles.at(m_currentIndex);
```
Comment (Clauszy) [inline customscreensaver/deepin-custom-screensaver/src/slideshowscreensaver.cpp:55]: 检查越界？

File: customscreensaver/deepin-custom-screensaver/src/slideshowscreensaver.cpp [hunk 4]
```diff
@@ -97,7 +91,63 @@ void SlideshowScreenSaver::paintEvent(QPaintEvent *event)
...
+
+    QPainter pa(&pip);
+    pa.setPen(Qt::white);
+    pa.drawText(pip.rect(), Qt::AlignCenter, tr("Picture not found")); // This text may be written in the configuration in the future
+
+    QPainter p(this);
+    p.drawPixmap(event->rect().topLeft(), pip, QRectF(QPointF(event->rect().topLeft()) * scale, QSizeF(event->rect().size()) * scale));
+}
+
+bool SlideshowScreenSaver::getCurrentImageIndex()
```
Comment (Clauszy) [inline customscreensaver/deepin-custom-screensaver/src/slideshowscreensaver.cpp:115]: get接口应该返回的是index值。
接口名与实现不一致

File: customscreensaver/deepin-custom-screensaver/src/slideshowscreensaver.cpp [hunk 5]
```diff
@@ -97,7 +91,63 @@ void SlideshowScreenSaver::paintEvent(QPaintEvent *event)
...
+    }
+}
+
+bool SlideshowScreenSaver::loadSlideshowImages()
+{
+    m_imagefiles.clear();
+    QString path = SlideshowConfig::instance()->slideshowPath();
+
+    QFileInfo fileInfo(path);
+    if (fileInfo.isDir() && fileInfo.exists()) {
```
Comment (Clauszy) [inline customscreensaver/deepin-custom-screensaver/src/slideshowscreensaver.cpp:136]: 应该先判断是否存在

**Review by Clauszy — CHANGES_REQUESTED** [review]

**Review by lvwujun — APPROVED** [review]

**Review by misonlan — APPROVED** [review]

**Review by shzhuangpub — APPROVED** [review]

**Review by Clauszy — APPROVED** [review]

**Pull request merged**